### DISCUSSION TITLE
Prestashop 8.1.7 search not returning results

### DISCUSSION THREAD

**Discussion by nicolasbd2** [PrestaShop/PrestaShop#37165] [Q&A] [answered]
Hi,
I've a PS 8.1.7 no PHP8.2 site and I have trouble getting the search to work. 
The database has search tables with data but no result found on the front end.
When I go to debug, I get the following error showing results are found but can't be displayed:
 
Key 'id_product' doesn't exist in table 'image_shop'<br /><br /><pre>SELECT p.*, product_shop.*, stock.out_of_stock, IFNULL(stock.quantity, 0) as quantity, pl.`description_short`, pl.`available_now`, pl.`available_later`, pl.`link_rewrite`, pl.`name`, image_shop.`id_image` id_image, il.`legend`, m.`name` manufacturer_name ,( SELECT SUM(weight) FROM scs_search_word sw LEFT JOIN scs_search_index si ON sw.id_word = si.id_word WHERE sw.id_lang = 2 AND sw.id_shop = 1 AND si.id_product = p.id_product AND (sw.word LIKE '%chaine%') ) position, DATEDIFF( p.`date_add`, DATE_SUB( "2024-10-16 00:00:00", INTERVAL 210 DAY ) ) > 0 new, product_attribute_shop.minimal_quantity AS product_attribute_minimal_quantity, IFNULL(product_attribute_shop.`id_product_attribute`,0) id_product_attribute FROM scs_product p INNER JOIN scs_product_shop product_shop ON (product_shop.id_product = p.id_product AND product_shop.id_shop = 1) INNER JOIN `scs_product_lang` pl ON ( p.`id_product` = pl.`id_product` AND pl.`id_lang` = 2 AND pl.id_shop = 1 ) LEFT JOIN `scs_product_attribute_shop` product_attribute_shop FORCE INDEX (id_product) ON (p.`id_product` = product_attribute_shop.`id_product` AND product_attribute_shop.`default_on` = 1 AND product_attribute_shop.id_shop=1) LEFT JOIN scs_stock_available stock ON (stock.id_product = `p`.id_product AND stock.id_product_attribute = 0 AND stock.id_shop = 1 AND stock.id_shop_group = 0 ) LEFT JOIN `scs_manufacturer` m FORCE INDEX (PRIMARY) ON m.`id_manufacturer` = p.`id_manufacturer` LEFT JOIN `scs_image_shop` image_shop FORCE INDEX (id_product) ON (image_shop.`id_product` = p.`id_product` AND image_shop.cover=1 AND image_shop.id_shop=1) LEFT JOIN `scs_image_lang` il ON (image_shop.`id_image` = il.`id_image` AND il.`id_lang` = 2) WHERE p.`id_product` IN (70,61,55,880,54,209,283,101,772,145,308,29,30,307,1147,143,31,310,1159,2938,2939,3084,146,123,28,309,25,124,892,1007,497,319,318,121,32,1165,320,3034,147,314,148,317,1155,1154,191,927,3132,532,198,265,212,312,56,3133,313,1269,2750,1270,2959,1840,1678,293,929,1626,759,296,761,760,1627,294,297,26,262,258,259,260,261,263,266,528,529,221,1783,526,527,58,59,281,288,3066,53,1536,1535,576,852,1250,1251,51,52,1166,1167,566,57,94,286,1455,375,1215,904,894,381,1329,1330,864,1934,2705,421,682,1529,423,1530,441,442,443,444,445,446,447,448,449,450,451,1022,495,648,490,647,1437,3050,792,592,669,762,1261,1581,961,960,958,959,573,962,805,806,1168,1281,1282,1283,1284,1285,1286,834,1122,197,821,1633,365,366,1356,170,2946,875,890,891,3236,895,1624,898,1623,642,643,896,897,1632,1196,1174,2963,1233,1234,1236,1235,1162,1669,1671,1670,1672,1668,2744,2745,2746,1205,1204,1206,534,1568,1569,1567,3033,520,530,521,519,1224,523,524,525,60,590,2878,3241,1625,175,176,173,1629,1970,3094,3096,3369,2921,2911,2904,2982,1066,3120,3121,1555,1121,1557,2805,3114,3115,3124,3126,3127,1558,3169,3171,3172,3142,3156,3168,2987,1612,1613,861,2537,2539,2844,2846,2847,2848,2849,2850) GROUP BY product_shop.id_product ORDER BY position desc LIMIT 0,100</pre>
at line 773 in file classes/db/Db.php

Any clue?
Thx a lot

**Comment by Hlavtox** [comment, marked as answer]
You have corrupted database structure. On 8.1.7, there is an id_product column in ps_image_shop.

**Comment by nicolasbd2** [comment]
Thanks for the answer, but I do have the column in the table.
![2024-10-17_104319](https://github.com/user-attachments/assets/51e333e0-a572-4ad4-b707-4d0a9234c1eb)
